### PR TITLE
cpm: init at 0.35.1

### PR DESCRIPTION
**Pull request by ken-matsui**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Related to: https://github.com/NixOS/nixpkgs/pull/178258#issuecomment-1181209078

@AndersonTorres, could you please review this PR at your convenience?

### CODE REVIEW

File: pkgs/development/tools/cpm/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
```
[the remark this answers is not in the document]
Comment (ken-matsui) [reply pkgs/development/tools/cpm/default.nix:6]: Thank you for your review! :)

**Comment by ken-matsui** [comment]
I will squash commit when this PR is ready.

**Comment by ken-matsui** [comment]
Thank you!

File: pkgs/development/tools/cpm/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    hash = "sha256-Oon/5iwkUUASsUDvde69iEwLe8/CAzwYKYsyzH5K+V0=";
+  };
+
+  dontConfigure = true;
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p ${placeholder "out"}/share/cpm/
+    cp ./cmake/CPM.cmake ${placeholder "out"}/share/cpm/
```
Comment (AndersonTorres) [inline pkgs/development/tools/cpm/default.nix:23]: Remember to decorate custom phases, surrounding the code with `runHook pre<phase name>` and  `runHook post<phase name>`.

Comment (ken-matsui) [reply pkgs/development/tools/cpm/default.nix:23]: Ah, thank you for pointing it out!

**Pull request merged**